flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/ClusterConfigurationInfo.java

```diff
@@ -45,10 +59,19 @@ public static ClusterConfigurationInfo from(Configuration config) {
 		final Map<String, String> configurationWithHiddenSensitiveValues = ConfigurationUtils.hideSensitiveValues(config.toMap());
 
 		for (Map.Entry<String, String> keyValuePair : configurationWithHiddenSensitiveValues.entrySet()) {
-			clusterConfig.add(new ClusterConfigurationInfoEntry(keyValuePair.getKey(), keyValuePair.getValue()));
+			clusterConfig.add(createClusterConfigurationInfoEntry(keyValuePair.getKey(), keyValuePair.getValue()));
 		}
 
 		return clusterConfig;
 	}
 
+	@VisibleForTesting
+	static ClusterConfigurationInfoEntry<?> createClusterConfigurationInfoEntry(String key, String valueStr) {
+		if (MEMORY_OPTION_KEYS.contains(key)) {
+			return new ClusterConfigurationInfoEntry<>(key, MemorySize.parse(valueStr).getBytes());
+		} else {
+			return new ClusterConfigurationInfoEntry<>(key, valueStr);
+		}
```

Comment:
I am not sure whether we are solving the problem correctly here. The problem I have with this solution is that we are adjusting the formatting of the `ClusterConfigHandler` because a user of this handler cannot parse the values properly. This does not seem right to me. Moreover, I would find it confusing to see a differently specified memory configuration in the web UI where the config is displayed.

I could see two solutions to the problem: 1) Add the parsing logic to the consumer of the data (namely the web UI). 2) Create an endpoint which serves the JM memory configuration. In this handler, I'd be ok with outputting the memory configuration as bytes.

Reply:
hm, thinking about it - is there a usecase for the `jobmanager/config` endpoint? ...since we're misusing it right now already by injecting the memory configuration and the `web.url` which are not set by the user but Flink itself.

Reply:
It is used for displaying the configuration in the web UI, right? Adding/enriching the memory configuration which are potentially derived from some other values is fine in my opinion.

From a maintenance point of view I would also say that this change might lead to confusion. Assuming time passes and somebody new has to change this part. I am pretty sure that he will not understand why certain options are reformatted and others not.

Reply:
And another clarification here: We're adjusting the memory configuration here already since we're transforming any memory setting from whatever unit the user used into bytes.

Anyway, it's good that you raised your concerns: I got back to @vthinkxie re-iterating over the problem ones more realizing that I misunderstood him. He's fine with doing the conversion on the client side (i.e. in the web UI). Hence, we're gonna go with your solution (1).